### PR TITLE
Add example extracting cities with name and population.

### PR DESCRIPTION
**Pull request by boxdot**

### CODE REVIEW

File: examples/cities.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#[derive(Debug, Default, Serialize)]
+struct City {
+    name: String,
+    population: usize,
+}
+
+/// Strings in an osmflat archive are stored in a single bytes blob separated by
+/// `0` character. Given an index, this function extracts a utf8 string starting
+/// at this index from the blob.
+fn get_str(strings: &[u8], start: usize) -> &str {
```
Comment (gferon) [inline examples/cities.rs:21]: I think we really need to move this to a helper in `osmflat`

**Review by gferon — APPROVED** [review]

File: examples/cities.rs (hunk 1 above)
Comment (boxdot) [reply examples/cities.rs:21]: Makes sense. I will try to solve #25 first, and then merge this example.

Comment (VeaaC) [reply examples/cities.rs:21]: or even in flatdata-rs

File: examples/cities.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    let end = strings[start..]
+        .iter()
+        .position(|&c| c == 0)
+        .expect("invalid string");
+    std::str::from_utf8(&strings[start..start + end]).expect("invalid string")
+}
+
+// Returns an iterator though all tags of a node given by its index in osmflat
+// archive.
+fn tags_for_node(archive: &Osm, node_idx: usize) -> impl Iterator<Item = (&str, &str)> + Clone {
```
Comment (VeaaC) [inline examples/cities.rs:31]: This also looks like a nice helper: Getting tags for nodes/ways

**Review by VeaaC — APPROVED** [review]

File: examples/cities.rs (hunk 1 above)
Comment (boxdot) [reply examples/cities.rs:21]: Don't we need a string block type in flatdata for that?

**Pull request merged**